python/ray/_private/client_mode_hook.py

```diff
@@ -75,13 +77,15 @@ def enable_client_mode():
         _explicitly_disable_client_mode()
 
 
-def client_mode_hook(func):
-    """Decorator for ray module methods to delegate to ray client"""
+def client_mode_hook(func=None, *, auto_init: bool = True):
```

Comment:
(1) Can you supply a comment on when `auto_init` should be set to `True` or `False`? When someone adds a new client API, how do they make a decision?
(2) Should `auto_init` not have a default vale, or default to False? It seems there are 10 `auto_init=True` and 4 `auto_init=False`. What happens when it is specified incorrectly?

Reply:
Great idea @mwtian ! I will remove the default value, I will also add some details in the `client_mode_hook` function. The main reason to use `False` is if the function:

1) The function does not need Ray to be initialized (i.e. `kv_store` functions)
2) Ray uses this function during the initialization path (i.e. `init`)

Reply:
Thanks for the explanations! Side question, does it mean if a function is used during the initialization path, then it cannot have auto init behavior?

Reply:
Correct!